### PR TITLE
#56944 Implment transfer budget to other user

### PR DESCRIPTION
**Pull request by LubomirMaron**

### CODE REVIEW

File: client/src/app/budgets/otherBudgetsDetail/otherBudgetsDetail.component.html [hunk 1]
```diff
@@ -2,25 +2,39 @@
     <h2>
         Budget Edit - {{budget.year}} - {{budget.user.firstName}} {{budget.user.lastName}}
     </h2>
+    <table>
```
Comment (menyet) [inline client/src/app/budgets/otherBudgetsDetail/otherBudgetsDetail.component.html]: Give it some style, this is how it looks like:

image

File: server/ERNI.PBA.Server.Host/Controllers/BudgetController.cs [hunk 2]
```diff
@@ -291,6 +297,33 @@ public async Task<IActionResult> UpdateBudget([FromBody] UpdateBudgetModel paylo
             return Ok();
         }
 
+        [HttpPut("{budgetId}/transfer/{userId}")]
+        [Authorize(Roles = Roles.Admin)]
+        public async Task<IActionResult> TransferBudget(int budgetId,int userId,CancellationToken cancellationToken)
```
Comment (menyet) [inline server/ERNI.PBA.Server.Host/Controllers/BudgetController.cs]: spaces

File: server/ERNI.PBA.Server.Host/Controllers/BudgetController.cs [hunk 3]
```diff
@@ -291,6 +297,33 @@ public async Task<IActionResult> UpdateBudget([FromBody] UpdateBudgetModel paylo
...
+        {
+            var budget = await _budgetRepository.GetBudget(budgetId, cancellationToken);
+            if (budget == null)
+            {
+                return BadRequest($"Budget with id {budgetId} not found");
+            }
+
+            if (!BudgetType.Types.First(type => type.Id == budget.BudgetType).IsTransferable)
```
Comment (menyet) [inline server/ERNI.PBA.Server.Host/Controllers/BudgetController.cs]: Single

Comment (LubomirMaron) [reply server/ERNI.PBA.Server.Host/Controllers/BudgetController.cs]: Good catch

File: server/ERNI.PBA.Server.Host/Controllers/BudgetController.cs [hunk 4]
```diff
@@ -51,16 +51,22 @@ public async Task<IActionResult> GetUserBudgetByYear(int userId, int year, Cance
         }
 
         [HttpGet("{budgetId}")]
-        [Authorize(Roles = Roles.Admin + "," + Roles.Viewer)]
-        public async Task<IActionResult> GetUserBudgetByYear(int budgetId, CancellationToken cancellationToken)
+        [Authorize]
```
Comment (menyet) [inline server/ERNI.PBA.Server.Host/Controllers/BudgetController.cs:54]: Maybe I asked this question already, but what about the viewer role?

Comment (LubomirMaron) [reply server/ERNI.PBA.Server.Host/Controllers/BudgetController.cs:54]: This endpoint is called only from otherBudgetsDetail.component. Thats route is admin only.  I dont think that check for viewer is needed

**Pull request merged**